cmd/manager/main.go

```diff
@@ -230,6 +226,13 @@ func ensureDefaults(cfg *rest.Config, scheme *kruntime.Scheme) error {
 
 func GenerateMetrics(cfg *rest.Config, namespace string) {
 
+	// convert the value of MetricsPort from string to int32.
+	port, err := strconv.ParseInt(customMetrics.MetricsPort, 10, 32)
+	if err != nil {
+		log.Errorf("Unable to parse metrics port: %v", err)
```

Comment:
If this fails, won't there be an issue when you call `metricsPort := int32(port)`?

Reply:
Actually no. It would return a zero when anything other than a valid integer is passed (like a string).